hw/mips/malta.c

```diff
@@ -1220,6 +1228,24 @@ static void mips_create_cpu(MachineState *ms, MaltaState *s,
     }
 }
 
+static void create_virtio_devices(void)
+{
+    CPUMIPSState *env;
+    MIPSCPU *cpu;
+
+    cpu = MIPS_CPU(first_cpu);
+    env = &cpu->env;
+    for(int i = VIRTIO_N_TRANSPORTS-1; i >= 0; i--) {
+        DeviceState* dev = sysbus_create_simple(TYPE_VIRTIO_MMIO,
+            VIRTIO_MMIO_MMAP_BASE+(VIRTIO_MMIO_MMAP_SIZE*i),
+            env->irq[VIRTIO_MMIO_IRQ_START+i]);
+
+        // This seems to breaking encapsulation a bit, but I cannot find an accessor...
+        VirtIOMMIOProxy *mmio = VIRTIO_MMIO(OBJECT(dev));
+        mmio->legacy = 0;
```

Comment:
I don't like forcing non-legacy VirtIO (FreeBSD is only just now ok with it, and no other bits of QEMU enable modern MMIO by default). There's a force-legacy property on the MMIO transport implementation, you want to configure QEMU to make that false. I haven't myself worked out how to do that from the command line, though.

Reply:
Default is now legacy